backend/src/recommitment/recommitment.service.ts

```diff
@@ -39,7 +38,118 @@ export class RecommitmentService {
   ) {
     this.logger.setContext(RecommitmentService.name);
   }
+  /**
+   * Update endpoint for recommitment reinitiation by coordinator.
+   * @param id
+   * @param recommitmentUpdate
+   * @returns
+   */
+  async coordinatorUpdatePersonnelRecommitment(
+    id: string,
+    recommitmentUpdate: PersonnelRecommitmentDTO,
+  ): Promise<PersonnelEntity> {
+    const key = Object.keys(recommitmentUpdate).includes(Program.EMCR)
+      ? Program.EMCR
+      : Program.BCWS;
+    await this.recommitmentRepository.update(
+      {
+        personnelId: id,
+        recommitmentCycleId: new Date().getFullYear(),
+        program: key,
+      },
+      {
+        status: recommitmentUpdate[key]?.status,
+        memberDecisionDate: null,
+        memberReason: null,
+        supervisorIdir: null,
+        supervisorReason: null,
+      },
+    );
+
+    const recommitmentCycle = await this.checkRecommitmentPeriod();
+
+    const dayAfterEndDate = addDays(recommitmentCycle.endDate, 1);
+    const currentPST = datePST(new Date());
+    const endDate =
+      isAfter(currentPST, dayAfterEndDate) &&
+      isAfter(currentPST, recommitmentCycle.startDate) &&
+      recommitmentCycle.reinitiationEndDate
+        ? recommitmentCycle.reinitiationEndDate
+        : recommitmentCycle.endDate;
+
+    const recommitment = await this.recommitmentRepository.findOneByOrFail({
+      personnelId: id,
+      recommitmentCycleId: new Date().getFullYear(),
+      program: key,
+    });
+
+    const emailTemplate = this.mailService.generateTemplate(
+      EmailTags.MEMBER_REACTIVATE,
```

Comment:
Don't love this. Makes the assumption of the status passed in. Same with the corresponding segment in the supervisor function.

I would add an if statement at the start of each function (or the controller endpoint), which throws an error if the update is not one of the expected statuses.